ui/pages/settings/flask/snaps-list-tab/view-snap/view-snap.component.js

```diff
@@ -0,0 +1,142 @@
+            <Box className="view-snap__pill-container" paddingLeft={2}>
+              <SnapsAuthorshipPill
+                packageName={snap.id}
+                url={snap.manifest.repository?.url}
+              />
+            </Box>
+            <Box
+              paddingLeft={4}
+              className="snap-settings-card__toggle-container view-snap__toggle-container"
+            >
+              <Tooltip interactive position="bottom" html={t('snapsToggle')}>
+                <ToggleButton
+                  value={snap.enabled}
+                  onToggle={() => onToggle(snap)}
+                  className="snap-settings-card__toggle-container__toggle-button"
+                />
+              </Tooltip>
+            </Box>
+          </Box>
+        </div>
+        <Box className="view-snap__content-container" width="7/12">
+          <div className="view-snap__section">
+            <Typography
+              variant={TYPOGRAPHY.H6}
+              color={COLORS.UI4}
+              boxProps={{ marginTop: 5 }}
+            >
+              {snap.manifest.description}
+            </Typography>
+          </div>
+          <div className="view-snap__section view-snap__permission-list">
+            <Typography variant={TYPOGRAPHY.H4}>{t('permissions')}</Typography>
+            <Typography variant={TYPOGRAPHY.H6} color={COLORS.UI4}>
+              {t('snapAccess', [snap.manifest.proposedName])}
+            </Typography>
+            <Box width="10/12">
+              <PermissionsConnectPermissionList
+                permissions={snap.manifest.initialPermissions}
+              />
+            </Box>
+          </div>
+          <div className="view-snap__section">
+            <Box width="11/12">
+              <Typography variant={TYPOGRAPHY.H4}>
+                {t('connectedSites')}
+              </Typography>
+              <Typography variant={TYPOGRAPHY.H6} color={COLORS.UI4}>
+                {t('connectedSnapSites', [snap.manifest.proposedName])}
+              </Typography>
+              <ConnectedSitesList
+                connectedSubjects={connectedSubjects}
+                onDisconnect={(origin) => {
+                  onDisconnect(origin, snap.permissionName);
+                }}
+              />
+            </Box>
+          </div>
+          <div className="view-snap__section">
+            <Typography variant={TYPOGRAPHY.H4}>{t('removeSnap')}</Typography>
+            <Typography
+              variant={TYPOGRAPHY.H6}
+              color={COLORS.UI4}
+              boxProps={{ paddingBottom: 3 }}
+            >
+              {t('removeSnapDescription')}
+            </Typography>
+            <Button
+              className="view-snap__remove__button"
+              type="danger"
+              css={{
+                maxWidth: '175px',
+              }}
+              onClick={() => {
+                dispatch(removeSnap(snap));
+                history.push(SNAPS_LIST_ROUTE);
+              }}
```

Comment:
Two issues here: first `removeSnap` is async, and we probably want to wait on that before redirecting so that the user doesn't still see their snap listed. But more importantly, `removeSnap` is not an action creator, so it should not be passed to `dispatch`. It's just a regular function.

```suggestion
              onClick={async () => {
                await removeSnap(snap);
                history.push(SNAPS_LIST_ROUTE);
              }}
```

Reply:
Actually... after trying this out, there was still one render of the old removed snap being listed, and it's my fault :facepalm:. The `forceUpdateMetamaskState` call would prevent this, but since that was removed, the state update doesn't propagate to the UI for a few milliseconds, leaving us 1 render cycle with stale data. Oops.

Not a big deal at all, as a user it's so quick it's not confusing in the slightest. Just not ideal.